pkg/services/alerting/eval_context.go

```diff
@@ -107,7 +107,7 @@ func (c *EvalContext) shouldUpdateAlertState() bool {
 
 // GetDurationMs returns the duration of the alert evaluation.
 func (c *EvalContext) GetDurationMs() float64 {
-	return float64(c.EndTime.Nanosecond()-c.StartTime.Nanosecond()) / float64(1000000)
+	return float64(c.EndTime.Sub(c.StartTime).Nanoseconds()) / float64(time.Millisecond)
```

Comment:
I am surprised the original code doesn't just do this:

```suggestion
	return float64(c.EndTime.Sub(c.StartTime).Milliseconds())
```

@grafana/alerting-squad?

Reply:
Just seems like a bug, George. That code is fairly old. If anyone knows why Chesterton's fence is there it's @berq

Reply:
Indeed, the only difference that I can think of when using `Milliseconds` directly is that the result would be always an `int64`. For instance from the small test in this PR:

```
=== RUN   TestGetDurationFromEvalContext
    eval_context_test.go:420:
        	Error Trace:	eval_context_test.go:420
        	Error:      	Not equal:
        	            	expected: 852.272
        	            	actual  : 852
        	Test:       	TestGetDurationFromEvalContext
--- FAIL: TestGetDurationFromEvalContext (0.00s)
```

Still, if the granularity of sub-millisecond measurements is desired/useful could be a valid question.

Reply:
I'd rather touch as little of the interface as possible, I think your float64 change is the minimum thing.